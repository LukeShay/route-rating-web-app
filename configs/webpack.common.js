const { resolve } = require('path');
const { CleanWebpackPlugin } = require('clean-webpack-plugin');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const { ProvidePlugin, EnvironmentPlugin } = require('webpack');

const SRC_DIR = resolve(__dirname, '../', 'src/');
const NODE_DIR = resolve(__dirname, '../', 'node_modules/');

module.exports = {
  resolve: {
    extensions: ['.ts', '.tsx', '.js', '.jsx'],
  },
  context: SRC_DIR,
  module: {
    rules: [
      {
        test: /\.js$/,
        use: ['babel-loader'],
        exclude: NODE_DIR,
        sideEffects: false,
      },
      {
        test: /\.tsx?$/,
        exclude: NODE_DIR,
        use: {
          loader: 'ts-loader?configFile=tsconfig.webpack.json',
        },
        sideEffects: false,
      },
      {
        test: /\.css$/,
        use: ['style-loader', 'css-loader'],
        sideEffects: false,
      },
      {
        test: /\.(jpe?g|png|gif|ico)$/i,
        loaders: [
          'file-loader?hash=sha512&digest=hex&name=img/[hash].[ext]',
          'image-webpack-loader?bypassOnDebug&optipng.optimizationLevel=7&gifsicle.interlaced=false',
        ],
      },
      {
        test: /\.svg$/,
        use: ['@svgr/webpack'],
      },
    ],
  },
  plugins: [
    new CleanWebpackPlugin(),
    new HtmlWebpackPlugin({
      template: 'index.html.ejs',
      favicon: 'favicon.ico',
    }),
    new ProvidePlugin({
      React: 'react',
    }),
    new EnvironmentPlugin(['recaptchaKey']),
  ],
  externals: {
    react: 'React',
    'react-dom': 'ReactDOM',
  },
};
