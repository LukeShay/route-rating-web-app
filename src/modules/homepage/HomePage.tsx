import Typography from '@material-ui/core/Typography';
import React from 'react';

const HomePage: React.FC = (): JSX.Element => {
  return (
    <React.Fragment>
      <Typography variant="h2">Climbing Routes Rating</Typography>
      <Typography variant="h3">
        Web application for gyms to post their routes and customers to rate
        them.
      </Typography>
    </React.Fragment>
  );
};

export default HomePage;
