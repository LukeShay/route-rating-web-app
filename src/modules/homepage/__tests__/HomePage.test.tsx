import React from "react";
import HomePage from "../HomePage";
import { shallow } from "../../../../configs/setupEnzyme";
import { shallowToJson } from "enzyme-to-json";

describe("<HomePage />", () => {
  it("should render correctly.", () => {
    expect(shallowToJson(shallow(<HomePage />))).toMatchSnapshot();
  });
});
