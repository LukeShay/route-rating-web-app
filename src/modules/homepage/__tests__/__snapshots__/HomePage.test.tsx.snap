// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render correctly. 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    variant="h2"
  >
    Climbing Routes Rating
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    variant="h3"
  >
    Web application for gyms to post their routes and customers to rate them.
  </WithStyles(ForwardRef(Typography))>
</Fragment>
`;
